### DISCUSSION TITLE
Does weblate support composite format string?

### DISCUSSION THREAD

**Discussion by steam3d** [WeblateOrg/weblate#5221] [Q&A] [answered]
I didn't found "composite format string" in Enforced checks. Maybe i something missed.
`"Name = {0}"`
I need {0} to be required in the translation string
https://docs.microsoft.com/ru-ru/dotnet/standard/base-types/composite-formatting#composite-format-string

**Comment by nijel** [comment]
We don't have support for this specific format,, but there is `c-sharp-format` which might cover it for most use cases. See https://docs.weblate.org/en/latest/user/checks.html#id1

**Comment by steam3d** [comment]
I tried it. It does nothing. I use the .Net resource file.
There is my settings and translation example:

![image](https://user-images.githubusercontent.com/30986874/104823940-fc6b6980-585e-11eb-93ce-4578f1ff8f7e.png)
![image](https://user-images.githubusercontent.com/30986874/104823959-26bd2700-585f-11eb-88ef-f1c539d155e1.png)

What i am trying to achieve

![image](https://user-images.githubusercontent.com/30986874/104823977-4f452100-585f-11eb-99ce-a88969a06f4e.png)

**Reply by nijel** [reply]
It needs to be enabled in the translation flags...

**Reply by steam3d** [reply]
Thx!

**Comment by steam3d** [comment, marked as answer]
A bit confusing that the same value can be specified in several places.
To enable it write to translation flags as answered @nijel 
![image](https://user-images.githubusercontent.com/30986874/105003165-5f1d5a80-5a43-11eb-8c23-175a9ce3e1fb.png)

**Reply by nijel** [reply]
Each of the configuration does a different thing. The enforced checks define the ones which can't be dismissed by the user. The flags define which checks to run on a string and these can be defined on string level as well. This distinction is intentional as you might want to enforce checks which are enabled only on some strings (for example rendering size checks).

Any suggestions on making this more intuitive are welcome.

**Reply by nijel** [reply]
Weblate 4.5 will indicate this problem with an alert:

![image](https://user-images.githubusercontent.com/212189/107240878-b56e2f80-6a2a-11eb-8821-ad312128da89.png)
